app/test/check_tx_test.go

```diff
@@ -248,8 +255,10 @@ func TestCheckTx(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			resp, err := testApp.CheckTx(&abci.RequestCheckTx{Type: tt.checkType, Tx: tt.getTx()})
-			require.NoError(t, err)
+			resp, err = testApp.CheckTx(&abci.RequestCheckTx{Type: tt.checkType, Tx: tt.getTx()})
+			if resp.Code == 0 {
+				require.NoError(t, err)
+			}
```

Comment:
we don't always assume there is no error, the error return is new and was causing some tests to fail but the correct response code was being returned.